src/Microsoft.AspNet.Mvc.Razor/LanguageViewLocationExpander.cs

```diff
@@ -21,36 +22,32 @@ namespace RazorWebSite
     /// </example>
     public class LanguageViewLocationExpander : IViewLocationExpander
     {
-        private const string ValueKey = "language";
-        private readonly Func<ActionContext, string> _valueFactory;
-
-        /// <summary>
-        /// Initializes a new instance of <see cref="LanguageViewLocationExpander"/>.
-        /// </summary>
-        /// <param name="valueFactory">A factory that provides tbe language to use for expansion.</param>
-        public LanguageViewLocationExpander(Func<ActionContext, string> valueFactory)
-        {
-            _valueFactory = valueFactory;
-        }
-
         /// <inheritdoc />
         public void PopulateValues(ViewLocationExpanderContext context)
         {
-            var value = _valueFactory(context.ActionContext);
```

Comment:
This needs to put the `CurrentUICulture` in context.Values under some well known key.